### PR TITLE
TINY-6533: Turn off parts of image upload that RTC handles itself

### PR DESCRIPTION
**Pull request by TheSpyder**
Related Ticket: TINY-6533

Description of Changes:
* No user-visible changes, so no changelog, just turning off code that mutates the DOM when RTC is active.
* RTC already handles image upload updates; we just had special code in our mutation blocker to allow these modifications to the view. They had no effect on the model.
* Turning off the unnecessary modifications allows us to remove those checks from our mutation blocker and prevent a class of mutations that could be performed without an error.

From what I can tell the image scanner should _never_ need to convert base64 to blob anymore - Spocke updated the parser to do it - but I'm not prepared to verify that right now.

Pre-checks:
* [x] ~Changelog entry added~
* [x] ~Tests have been added (if applicable)~
* [x] Branch prefixed with `feature/` for new features (if applicable)

Review:
* [x] Milestone set
* [x] Review comments resolved

GitHub issues (if applicable):

### CODE REVIEW

**Review by ashfordneil — APPROVED** [review]
Just want to make sure that there's testing on the RTC side of things to make sure this works?

File: modules/tinymce/src/core/main/ts/api/EditorUpload.ts [hunk 1]
```diff
@@ -239,11 +243,14 @@ const EditorUpload = (editor: Editor): EditorUpload => {
         return true;
       });
 
-      Arr.each(result, (resultItem) => {
-        replaceUrlInUndoStack(resultItem.image.src, resultItem.blobInfo.blobUri());
-        resultItem.image.src = resultItem.blobInfo.blobUri();
-        resultItem.image.removeAttribute('data-mce-src');
-      });
+      if (!Rtc.isRtc(editor)) {
```
Comment (JamesToohey) [inline modules/tinymce/src/core/main/ts/api/EditorUpload.ts:246]: Is there any reason this one does a not condition and the others have the if/else? FWIW I'd prefer it this way

Comment (TheSpyder) [reply modules/tinymce/src/core/main/ts/api/EditorUpload.ts:246]: because I did them at different times 🤷‍♂️ will fix

**Comment by TheSpyder** [comment]
> Just want to make sure that there's testing on the RTC side of things to make sure this works?

yes, I'll link it in slack since I know we don't like private links in these discussions

**Review by lnewson — APPROVED** [review]

**Review by HAFRMO — APPROVED** [review]

**Pull request merged**